xonsh/tools.py

```diff
@@ -228,8 +228,28 @@ def __len__(self):
     def __delitem__(self, key):
         self._l.__delitem__(key)
 
+    @staticmethod
+    def _prepare_path(p):
+        return str(expand_path(p))
+
+    @staticmethod
+    def _check_path(p):
+        if not pathlib.Path(p).exists():
+            print(f"EnvPath warning: path {repr(p)} does not exists.", file=sys.stderr)
```

Comment:
It is not needed to check by default. Many of the times, the dirs in $PATH would have actually been deleted or yet to be created.